### PR TITLE
refactor test to jest and add jest-html-reporter

### PR DESCRIPTION
**Pull request by renardete**
image
image

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ env:
   es6: true
   node: true
   mocha: true
```
Comment (leonleo997) [inline .eslintrc.yml]: if you are not using mocha please delete it

File: jest.config.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    [
+      "jest-html-reporters",
+      {
+        "publicPath": "./report",
+        "filename": "ApiTest.html",
+        "expand": true
+      }
+    ]
+  ]
+}
```
Comment (leonleo997) [inline jest.config.json]: EOF

File: package.json [hunk 3]
```diff
@@ -31,6 +31,8 @@
     "eslint-config-airbnb-base": "^13.2.0",
     "eslint-plugin-import": "^2.18.0",
     "http-status-codes": "^1.3.2",
+    "jest": "^24.8.0",
+    "jest-html-reporters": "^1.1.8",
     "mocha": "^6.1.4"
```
Comment (leonleo997) [inline package.json]: delete useless dependencies

File: test/MyFirstApiConsume.test.js [hunk 4]
```diff
@@ -1,13 +1,12 @@
 const agent = require('superagent-promise')(require('superagent'), Promise);
```
Comment (leonleo997) [inline test/MyFirstApiConsume.test.js]: Why is this structure?

Comment (renardete) [reply test/MyFirstApiConsume.test.js]: This is a function which require two parameter to return the actual wrapper to make the HTTP requests.

**Pull request merged**